internal/core/operations/test_operation/test_operation_executor.go

```diff
@@ -0,0 +1,68 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package test_operation
+
+import (
+	"context"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	"go.uber.org/zap"
+)
+
+type TestOperationExecutor struct{}
+
+func NewExecutor() *TestOperationExecutor {
+	return &TestOperationExecutor{}
+}
+
+func (e *TestOperationExecutor) Execute(ctx context.Context, _op *operation.Operation, definition operations.Definition) error {
+	testOperationDefinition := definition.(*TestOperationDefinition)
+
+	zap.L().Sugar().Infof("sleeping routine for %d seconds", testOperationDefinition.SleepSeconds)
+
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
Any specific reason for using a ticker here? We could remove the necessity of the `for` below by using a select with a timer:

```go
select {
case <-time.After(time.Second * testOperationDefinition.SleepSeconds):
case <-ctx.Done():
	return ctx.Err()
}
```

The `select` will block the execution until it receives something in any of the channels.

Reply:
I need the context to be evaluated every second. So I need this sleep to look (second by second) to ctx.Done() channel